testsuite/MDAnalysisTests/coordinates/test_xdr.py

```diff
@@ -755,25 +764,34 @@ def test_nonexistent_offsets_file(self, traj):
         outfile_offsets = XDR.offsets_filename(traj)
         with patch.object(np, "load") as np_load_mock:
             np_load_mock.side_effect = IOError
-            saved_offsets = XDR.read_numpy_offsets(outfile_offsets)
-            assert_equal(saved_offsets, False)
+            with pytest.warns(UserWarning, match=re.escape(
+                    f"Failed to load offsets file {outfile_offsets}")):
+                saved_offsets = XDR.read_numpy_offsets(outfile_offsets)
+        assert saved_offsets == False
 
-    def test_nonexistent_offsets_file(self, traj):
+    def test_corrupted_offsets_file(self, traj):
         # assert that a corrupted file returns False during read-in
         # Issue #3230
         outfile_offsets = XDR.offsets_filename(traj)
         with patch.object(np, "load") as np_load_mock:
             np_load_mock.side_effect = ValueError
-            saved_offsets = XDR.read_numpy_offsets(outfile_offsets)
-            assert_equal(saved_offsets, False)
+            with pytest.warns(UserWarning, match=re.escape(
+                    f"Failed to load offsets file {outfile_offsets}")):
+                saved_offsets = XDR.read_numpy_offsets(outfile_offsets)
+        assert saved_offsets == False
 
     def test_reload_offsets_if_offsets_readin_io_fails(self, trajectory):
         # force the np.load call that is called in read_numpy_offsets
         # during _load_offsets to give an IOError
         # ensure that offsets are then read-in from the trajectory
         with patch.object(np, "load") as np_load_mock:
             np_load_mock.side_effect = IOError
-            trajectory._load_offsets()
+            with pytest.warns(UserWarning) as record:
+                trajectory._load_offsets()
+            assert len(record) == 2
```

Comment:
checking the length of the record is very flaky, can we instead just check that the two warnings you're looking for are in there? "X in Y" type of thing

Reply:
ok, added in [4f18afb](https://github.com/MDAnalysis/mdanalysis/pull/4231/commits/4f18afbb8e423386daf600cd6b13e56158e91eeb)